staging/src/k8s.io/apimachinery/pkg/util/sets/set.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sets
+
+import (
+	"sort"
+)
+
+// SetTypes is constraints on the types of elements of the set
+type SetTypes interface {
+	comparable
+}
```

Comment:
what's the value of this type?

Reply:
I think you mean "maybe remove this type and just use `comparable` directly"? Seems reasonable to me unless the documentation is extremely helpful

Reply:
Yeah, that's what I meant.